zeroex/order.go

```diff
@@ -215,7 +223,21 @@ func (s *SignedOrder) ConvertToOrderWithoutExchangeAddress() wrappers.OrderWitho
 	return orderWithoutExchangeAddress
 }
 
-// SignOrder converts an order to a signed 0x order
+// SignTestOrder converts a test 0x order to a signed 0x order
+func SignTestOrder(order *Order) (*SignedOrder, error) {
```

Comment:
One reason I wanted to use a `Signer` interface is so that we could simplify the code for `SignOrder`. This allows us to abstract away the parts of the code that are the same (generating an order hash, signing it, adding the `Signature` field to the order) and the parts that are different (*how* exactly the order hash is signed).

What do you think of changing the API for `SignOrder` to:

```go
func SignOrder(signer ethereum.Signer, order Order) (*SignedOrder, error)
```